x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/AbstractConvertFunction.java

```diff
@@ -43,7 +43,7 @@ protected Supplier<EvalOperator.ExpressionEvaluator> evaluator(Supplier<EvalOper
         DataType sourceType = field().dataType();
         var evaluator = evaluators().get(sourceType);
         if (evaluator == null) {
-            throw new AssertionError("unsupported type [" + sourceType + "]");
+            throw new UnsupportedOperationException("unsupported type [" + sourceType + "]");
```

Comment:
We have some QL Exceptions, that split in sever and client and `EsqlIllegalArgumentException` might be a good choice for these ones.

Reply:
This looks more appropriate to me, as it seems to be intended for server-side errors that should've been handled during validation.

@luigidellaquila suggested `UnsupportedOperationException` [here](https://github.com/elastic/elasticsearch/pull/98352#pullrequestreview-1580262191), but if we're all happy with `EsqlIllegalArgumentException` I'm going to go by your suggestion instead.